frontend/linux/rpm/distro/distro.go

```diff
@@ -0,0 +1,83 @@
+package distro
+
+import (
+	"context"
+	"path/filepath"
+
+	"github.com/Azure/dalec"
+	"github.com/Azure/dalec/frontend"
+	"github.com/Azure/dalec/frontend/linux"
+	"github.com/moby/buildkit/client/llb"
+	gwclient "github.com/moby/buildkit/frontend/gateway/client"
+	"github.com/moby/buildkit/frontend/subrequests/targets"
+)
+
+type Config struct {
+	FullName   string
+	ImageRef   string
+	ContextRef string
+
+	// The release version of the distro
+	ReleaseVer string
+
+	// Build dependencies needed
+	BuilderPackages []string
+
+	// Dependencies to install in base image
+	BasePackages       []string
+	RepoPlatformConfig *dalec.RepoPlatformConfig
+
+	DefaultOutputImage string
+
+	InstallFunc PackageInstaller
+
+	// Unique identifier for the package cache for this particular distro,
+	// e.g., azlinux3-tdnf-cache
+	CacheName string
+
+	// e.g. /var/cache/tdnf or /var/cache/dnf
+	CacheDir string
+}
+
+func (cfg *Config) PackageCacheMount(root string) llb.RunOption {
+	return llb.AddMount(filepath.Join(root, cfg.CacheDir), llb.Scratch(), llb.AsPersistentCacheDir(cfg.CacheName, llb.CacheMountLocked))
+}
+
+func (c *Config) Install(pkgs []string, opts ...DnfInstallOpt) llb.RunOption {
+	var cfg dnfInstallConfig
+	dnfInstallOptions(&cfg, opts)
+
+	return dalec.WithRunOptions(c.InstallFunc(&cfg, c.ReleaseVer, pkgs), c.PackageCacheMount(cfg.root))
+}
+
+func (cfg *Config) Handle(ctx context.Context, client gwclient.Client) (*gwclient.Result, error) {
+	var mux frontend.BuildMux
+
+	mux.Add("rpm", linux.HandlePackage(cfg), &targets.Target{
```

Comment:
We can use the root linux package target handler for this, as well as the container